File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/inline/comment/direct/Java/C++/s863488345_commented.cpp
#include <iostream>
#include <vector>
#include <string>
using namespace std;

int mapNum;
int h, w;
vector<vector<char>> battleTown;
int bomberActNum;
string bomberAct;
int tank[2];
int xOfTank, yOfTank, xOfBomber, yOfBomber, s, k, a, b, x, y;

void moveBomber(int a, int b) {
    if (0 <= a && a <= h && 0 <= b && b <= w) {
        if (battleTown[a][b] == '^') {
            for (int i = a - 1; i >= 0; --i) {
                if (a <= 0) return;
                else {
                    if (battleTown[a-1][b] == '*') {
                        battleTown[a-1][b] = '.';
                        return;
                    } else if (battleTown[a-1][b] == '#') {
                        return;
                    } else {
                        a--;
                        if (a == 0) return;
                    }
                }
            }
        } else if (battleTown[a][b] == 'v') {
            for (int i = a + 1; i < h; ++i) {
                if (battleTown[a+1][b] == '*') {
                    battleTown[a+1][b] = '.';
                    return;
                } else if (battleTown[a+1][b] == '#') {
                    return;
                } else {
                    a++;
                    if (a == h) return;
                }
            }
        } else if (battleTown[a][b] == '<') {
            for (int i = b - 1; i >= 0; --i) {
                if (battleTown[a][b-1] == '*') {
                    battleTown[a][b-1] = '.';
                    return;
                } else if (battleTown[a][b-1] == '#') {
                    return;
                } else {
                    b--;
                    if (b == 0) return;
                }
            }
        } else if (battleTown[a][b] == '>') {
            for (int i = b + 1; i < w; ++i) {
                if (battleTown[a][b+1] == '*') {
                    battleTown[a][b+1] = '.';
                    return;
                } else if (battleTown[a][b+1] == '#') {
                    return;
                } else {
                    b++;
                    if (b == w) return;
                }
            }
        }
    }
}

int main() {
    ios::sync_with_stdio(false);
    cin >> mapNum;
    for (int i = 0; i < mapNum; ++i) {
        if (i > 0 && i < mapNum) {
            cout << endl;
        }
        cin >> h >> w;
        battleTown.resize(h, vector<char>(w));
        x = -1; y = -1;
        for (int j = 0; j < h; ++j) {
            string line;
            cin >> line;
            for (int k = 0; k < w; ++k) {
                battleTown[j][k] = line[k];
                if (battleTown[j][k] == '^' || battleTown[j][k] == 'v' || 
                    battleTown[j][k] == '<' || battleTown[j][k] == '>') {
                    x = j;
                    y = k;
                }
            }
        }
        cin >> bomberActNum >> bomberAct;
        if (0 <= x && x < h && 0 <= y && y < w) {
            for (int l = 0; l < bomberActNum; ++l) {
                char action = bomberAct[l];
                if (action == 'U') {
                    battleTown[x][y] = '^';
                    if (x - 1 >= 0 && battleTown[x-1][y] == '.') {
                        battleTown[x-1][y] = '^';
                        battleTown[x][y] = '.';
                        x--;
                    }
                } else if (action == 'D') {
                    battleTown[x][y] = 'v';
                    if (x + 1 < h && battleTown[x+1][y] == '.') {
                        battleTown[x+1][y] = 'v';
                        battleTown[x][y] = '.';
                        x++;
                    }
                } else if (action == 'L') {
                    battleTown[x][y] = '<';
                    if (y - 1 >= 0 && battleTown[x][y-1] == '.') {
                        battleTown[x][y-1] = '<';
                        battleTown[x][y] = '.';
                        y--;
                    }
                } else if (action == 'R') {
                    battleTown[x][y] = '>';
                    if (y + 1 < w && battleTown[x][y+1] == '.') {
                        battleTown[x][y+1] = '>';
                        battleTown[x][y] = '.';
                        y++;
                    }
                } else if (action == 'S') {
                    moveBomber(x, y);
                }
            }
            for (s = 0; s < h; ++s) {
                for (k = 0; k < w; ++k) {
                    cout << battleTown[s][k];
                }
                cout << endl;
            }
        }
    }
    return 0;
}
// End of Code

File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/direct/javadoc/comment/direct/Java/Go/s863488345_commented.go
package main

import (
	"fmt"
)

var (
	mapNum     int
	h, w       int
	battleTown [][]rune
	bomberActNum int
	bomberAct    []rune
	tank         [2]int
	xOfTank, yOfTank int
	xOfBomber, yOfBomber int
)

func main() {
	var n int
	fmt.Scan(&mapNum)

	for i := 0; i < mapNum; i++ {
		if i > 0 {
			fmt.Println()
		}
		fmt.Scan(&h, &w)
		battleTown = make([][]rune, h)
		for j := range battleTown {
			var line string
			fmt.Scan(&line)
			battleTown[j] = []rune(line)
			for k, char := range battleTown[j] {
				if char == '^' || char == 'v' || char == '<' || char == '>' {
					tank[0], tank[1] = j, k
				}
			}
		}

		fmt.Scan(&bomberActNum)
		var act string
		fmt.Scan(&act)
		bomberAct = []rune(act)

		for _, action := range bomberAct {
			switch action {
			case 'U':
				battleTown[tank[0]][tank[1]] = '^'
				if tank[0]-1 >= 0 && battleTown[tank[0]-1][tank[1]] == '.' {
					battleTown[tank[0]-1][tank[1]] = '^'
					battleTown[tank[0]][tank[1]] = '.'
					tank[0]--
				}
			case 'D':
				battleTown[tank[0]][tank[1]] = 'v'
				if tank[0]+1 < h && battleTown[tank[0]+1][tank[1]] == '.' {
					battleTown[tank[0]+1][tank[1]] = 'v'
					battleTown[tank[0]][tank[1]] = '.'
					tank[0]++
				}
			case 'L':
				battleTown[tank[0]][tank[1]] = '<'
				if tank[1]-1 >= 0 && battleTown[tank[0]][tank[1]-1] == '.' {
					battleTown[tank[0]][tank[1]-1] = '<'
					battleTown[tank[0]][tank[1]] = '.'
					tank[1]--
				}
			case 'R':
				battleTown[tank[0]][tank[1]] = '>'
				if tank[1]+1 < w && battleTown[tank[0]][tank[1]+1] == '.' {
					battleTown[tank[0]][tank[1]+1] = '>'
					battleTown[tank[0]][tank[1]] = '.'
					tank[1]++
				}
			case 'S':
				xOfBomber, yOfBomber = tank[0], tank[1]
				moveBomber(xOfBomber, yOfBomber)
			}
		}

		for _, row := range battleTown {
			for _, cell := range row {
				fmt.Printf("%c", cell)
			}
			fmt.Println()
		}
	}
}

func moveBomber(a, b int) {
	if a < 0 || a >= h || b < 0 || b >= w {
		return
	}

	switch battleTown[a][b] {
	case '^':
		for i := a - 1; i >= 0; i-- {
			if battleTown[i][b] == '*' {
				battleTown[i][b] = '.'
				return
			} else if battleTown[i][b] == '#' {
				return
			}
		}
	case 'v':
		for i := a + 1; i < h; i++ {
			if battleTown[i][b] == '*' {
				battleTown[i][b] = '.'
				return
			} else if battleTown[i][b] == '#' {
				return
			}
		}
	case '<':
		for i := b - 1; i >= 0; i-- {
			if battleTown[a][i] == '*' {
				battleTown[a][i] = '.'
				return
			} else if battleTown[a][i] == '#' {
				return
			}
		}
	case '>':
		for i := b + 1; i < w; i++ {
			if battleTown[a][i] == '*' {
				battleTown[a][i] = '.'
				return
			} else if battleTown[a][i] == '#' {
				return
			}
		}
	}
}

// End of Code
